# Generated by Django 2.2.1 on 2019-07-31 05:25

from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    initial = True

    dependencies = [
    ]

    operations = [
        migrations.CreateModel(
            name='LateRule',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('late_limit', models.IntegerField(blank=True, default=0)),
            ],
        ),
        migrations.CreateModel(
            name='RuleException',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(default='Not Specified', max_length=255)),
                ('start_date', models.DateField()),
                ('end_date', models.DateField()),
                ('office_start_time', models.TimeField(null=True)),
                ('office_end_time', models.TimeField(null=True)),
            ],
        ),
        migrations.CreateModel(
            name='OfficeRules',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(max_length=255)),
                ('office_start_time', models.TimeField(null=True)),
                ('office_end_time', models.TimeField(null=True)),
                ('active', models.BooleanField(default=False)),
                ('exceptions', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='rules.RuleException')),
                ('late_limit', models.OneToOneField(null=True, on_delete=django.db.models.deletion.CASCADE, to='rules.LateRule')),
            ],
        ),
    ]
